### PR TITLE
New `periodic-service-account` for periodic Prow jobs

### PR DESCRIPTION
**Pull request by jljaco**
Issue #, if available:  https://github.com/aws-controllers-k8s/community/issues/1577

Description of changes:
Define a new service account `periodic-service-account` with more limited IAM permissions (only S3 bucket access needed), and switch all current periodic Prow jobs over to using it. 

Also, make the `label_sync` job use the new GH app style auth mechanism (change its command line params).

Validation:
Example log output of a `label_sync` job that ran without errors:  https://prow-staging.ack.aws.dev/log?job=label-sync&id=1622752518511529984

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jaypipes — APPROVED** [review]
:+1: looks kosher to me, thanks @jljaco :)

**Review by A-Hilaly — APPROVED** [review]
:+1:

File: prow/jobs/jobs.yaml [hunk 1]
```diff
@@ -19,8 +20,9 @@ periodics:
       - --config=/etc/config/labels.yaml
       - --confirm=true
       - --orgs=aws-controllers-k8s
-      - --token=/etc/github/token
-      - --github-endpoint=http://ghproxy
+      - --github-app-id=260409
+      - --github-app-private-key-path=/etc/github/cert
+      - --github-endpoint=http://ghproxy.prow.svc.cluster.local
```
Comment (jljaco) [inline prow/jobs/jobs.yaml:25]: necessary because prowjobs run in the `test-pods` namespace

File: prow/jobs/jobs.yaml [hunk 2]
```diff
@@ -19,8 +20,9 @@ periodics:
       - --config=/etc/config/labels.yaml
       - --confirm=true
       - --orgs=aws-controllers-k8s
-      - --token=/etc/github/token
-      - --github-endpoint=http://ghproxy
+      - --github-app-id=260409
```
Comment (jljaco) [inline prow/jobs/jobs.yaml:23]: cleared w/ OSPO that this is OK to commit -- not a secret

File: prow/jobs/jinja/periodics/label_sync.jinja2 [hunk 3]
```diff
@@ -16,8 +17,9 @@
       - --config=/etc/config/labels.yaml
       - --confirm=true
       - --orgs=aws-controllers-k8s
-      - --token=/etc/github/token
-      - --github-endpoint=http://ghproxy
+      - --github-app-id=260409
```
Comment (A-Hilaly) [inline prow/jobs/jinja/periodics/label_sync.jinja2:20]: very tiny nit: in the future we can consider putting this value in a configmap and pass it as env var

**Review by A-Hilaly — APPROVED** [review]
LGTM, happy to merge this once @RedbackThomson comment is addressed

File: docs/prow-jobs.md [hunk 4]
```diff
@@ -11,7 +11,8 @@ We break down jobs into "presubmit", "postsubmit" and "periodic" types.
 "Presubmit" jobs must pass before a PR can be merged and are triggered by an 
 `/ok-to-test` command in the PR comments. "Postsubmit" jobs are triggered by the
 merging of a pull request and are typically used for publishing artifacts.
-"Periodic" jobs run on a set interval and are currently not used by ACK.
+"Periodic" jobs run on a set interval and are currently used to update the
```
[the remark this answers is not in the document]
Comment (jljaco) [reply docs/prow-jobs.md:14]: added wording to that effect

File: prow/jobs/jinja/periodics/label_sync.jinja2 (hunk 3 above)
Comment (jljaco) [reply prow/jobs/jinja/periodics/label_sync.jinja2:20]: certainly a good idea for the future

**Review by RedbackThomson — APPROVED** [review]
Thank you @jljaco 
/lgtm

**Pull request merged**